src/departments/IDepartment.java

```diff
@@ -0,0 +1,23 @@
+package departments;
+
+import java.util.List;
+
+import workers.Team;
+import bwapi.Unit;
+
+public interface IDepartment {
+  public enum Purpose{
+    GAMESTART,
+    MINE,
+    SCOUT,
+    BUILD,
+    DEFEND,
+    ATTACK,
+    CRITICAL
```

Comment:
Presumably, something that is critical to the victory condition and is given the highest priority i.e. pull workers from mining to defend against zerg rush, or sending all units to enemy base guarantees victory